h2o-logger/src/main/java/water/logging/LoggerFactory.java

```diff
@@ -0,0 +1,54 @@
+package water.logging;
+
+
+
+public class LoggerFactory {
+    private static final String DEFAULT_CLASS_TO_CHECK = "org.slf4j.LoggerFactory";
+    private static final LoggerFactory INSTANCE = new LoggerFactory(DEFAULT_CLASS_TO_CHECK);
+    
+    private String classToCheck;
+
+    LoggerFactory(String slf4jClass) {
+        classToCheck = slf4jClass;
+    }
+    
+    /**
+     * Returns new logger for each invocation. Logger logs to console / SLF4J (custom wrapper around SLF4J) depending on whether SLF4J is available.
+     *
+     * @param clazz class from which getLogger() is called
+     * @return Slf4JLogger if SLF4J is on the classpath, ConsoleLogger if not
+     */
+    public static Logger getLogger(Class<?> clazz) {
+        return INSTANCE.getCustomLogger(clazz);
+    }
+
+    
+    /**
+     * Returns new logger for each invocation.  Logger logs to console / SLF4J (custom wrapper around SLF4J) depending on whether SLF4J is available.
+     *
+     * @param clazz class from which getLogger() is called
+     * @return Slf4JLogger if SLF4J is on the classpath, ConsoleLogger if not
+     */
+    public Logger getCustomLogger(Class<?> clazz) {
+        if (isSlf4JAvailable()) {
```

Comment:
I think we should do this check only once at init-time, doing class.forName for every time a logger is created can be expensive I think